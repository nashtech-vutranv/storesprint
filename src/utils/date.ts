import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
