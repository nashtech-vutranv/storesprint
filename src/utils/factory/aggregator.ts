import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
