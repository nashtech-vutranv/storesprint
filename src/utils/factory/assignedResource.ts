import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
