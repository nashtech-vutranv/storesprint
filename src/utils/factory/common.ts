import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
