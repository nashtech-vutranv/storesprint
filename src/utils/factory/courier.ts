import faker from '@faker-js/faker'
import {ICourierMapping} from '../../interface'

export const mockCourier = (): ICourierMapping => {
  return {
    id: faker.datatype.string(),
    createdAt: faker.datatype.string(),
    modifiedAt: faker.datatype.string(),
    mmsValue: faker.datatype.string(),
    version: faker.datatype.number(),
    mappingTarget: {
        id: faker.datatype.string(),
        mappingTypeId: faker.datatype.string(),
        marketplaceCode: faker.datatype.string(),
        marketplaceTypeId: faker.datatype.string(),
        marketplaceValue: faker.datatype.string(),
    },
    mappingType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    mappingTypeId: faker.datatype.string(),
    marketplaceType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    marketplaceTypeId: faker.datatype.string(),
    marketplaceValue: faker.datatype.string(),
  }
}

export const mockCouriers = (): ICourierMapping[] => {
    return Array.apply(null, Array(10)).map(() =>mockCourier()) 
}
