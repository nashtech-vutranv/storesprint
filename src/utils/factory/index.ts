export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
