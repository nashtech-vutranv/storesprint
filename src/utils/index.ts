export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
