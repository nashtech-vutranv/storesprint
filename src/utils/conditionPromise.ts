import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}