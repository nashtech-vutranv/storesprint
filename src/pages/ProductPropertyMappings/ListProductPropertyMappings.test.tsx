import {render, fireEvent, waitFor} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {mockListProductPropertyMappings, mockOption} from '../../utils/factory'
import {ListProductPropertyMappings} from './'

const API = require('../../services/ProductPropertyMappingsService')

jest.mock(API, () => {
  const instance = jest.fn().mockReturnValue({
    filterProductPropertyMappings: jest
      .fn()
      .mockImplementation(() => Promise.resolve({
        data: mockListProductPropertyMappings()
      })),
  })
  return instance
})

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    productCategoryMappingsId: 'productCategoryMappingsId',
    marketplaceTypeId: 'marketplaceTypeId',
    mappingTargetId:'mappingTargetId'
}),
}))

beforeEach(() => {
  window.history.pushState({
    selectedMarketplaceType: mockOption(),
    selectedMarketplaceProductCategory: mockOption(),
    eventKey: '0',
    currentStatus: null
  }, '', '')
})

test('renders List product property mappings Page', async () => {
  const tree = render(
    <MemoryRouter>
      <ListProductPropertyMappings />
    </MemoryRouter>
  )
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
  
  const refresh = tree.container.querySelector(
    '.p-button-icon.pi-refresh'
  ) as Element
  await waitFor(() => {
    expect(refresh).toBeTruthy()
  })
  
  fireEvent.click(refresh)
  await waitFor(() =>
    expect(
      tree.queryByText(/product_property_mappings_filter_marketplace_type_label/i)
    ).toBeInTheDocument()
  )

  await waitFor(() => {
    expect(
      tree.queryByText(
        /product_property_mappings_filter_marketplace_type_placeHolder/i
      )
    ).toBeInTheDocument()
  })
  
  await waitFor(() =>
    expect(
      tree.queryByText(/product_property_mappings_filter_marketplace_product_category_label/i)
    ).toBeInTheDocument()
  )
  await waitFor(() => {
    expect(
      tree.queryByText(
        /product_property_mappings_filter_marketplace_product_category_placeHolder/i
      )
    ).toBeInTheDocument()
  })
})
