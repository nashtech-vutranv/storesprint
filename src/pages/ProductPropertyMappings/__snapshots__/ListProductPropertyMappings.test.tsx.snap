// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List product property mappings Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <nav
          aria-label="breadcrumb"
        >
          <ol
            class="breadcrumb m-0"
          >
            <li
              class="breadcrumb-item"
            >
              <a
                href="/product-category-mappings"
              >
                <span>
                  product_category_mappings_page_title
                </span>
              </a>
            </li>
            <li
              aria-current="page"
              class="breadcrumb-item active"
            >
              <span>
                 
                product_property_mappings_page_title
                 
              </span>
            </li>
          </ol>
        </nav>
        <div
          class="col"
        >
          <div
            class="page-title-box"
          >
            <h4
              class="page-title"
            >
              product_property_mappings_page_title
            </h4>
          </div>
        </div>
        <div
          class="col-12"
        >
          <div
            class="card"
          >
            <div
              class="card-body"
            >
              <div
                class="custom-accordion accordion"
                id="accordion"
              >
                <div
                  class="d-flex align-items-center"
                >
                  <span
                    class=""
                    style="margin-right: 0.5rem;"
                  >
                    <span>
                      <div
                        class="d-flex align-items-start"
                      >
                        <i
                          class="mdi mdi-chevron-down d-none d-sm-inline-block align-middle"
                          style="transform: scale(1.5);"
                        />
                      </div>
                    </span>
                  </span>
                  <span
                    style="font-size: 1.2rem; margin: 0px;"
                  >
                    product_property_mappings_filter_title
                  </span>
                </div>
                <div
                  class="accordion-collapse collapse show"
                >
                  <div
                    class="px-0 container-fluid"
                  >
                    <div
                      class="align-items-center pb-2 row"
                    >
                      <div
                        class="col-3"
                      >
                        <label
                          for="marketplace-type"
                        >
                          product_property_mappings_filter_marketplace_type_label
                        </label>
                      </div>
                      <div
                        class="col-6"
                      >
                        <div
                          class="react-select css-b62m3t-container"
                        >
                          <span
                            class="css-1f43avz-a11yText-A11yText"
                            id="react-select-2-live-region"
                          />
                          <span
                            aria-atomic="false"
                            aria-live="polite"
                            aria-relevant="additions text"
                            class="css-1f43avz-a11yText-A11yText"
                          />
                          <div
                            class="react-select__control css-13cymwt-control"
                          >
                            <div
                              class="react-select__value-container css-1fdsijx-ValueContainer"
                            >
                              <div
                                class="react-select__placeholder css-1jqq78o-placeholder"
                                id="react-select-2-placeholder"
                              >
                                product_property_mappings_filter_marketplace_type_placeHolder
                              </div>
                              <div
                                class="react-select__input-container css-qbdosj-Input"
                                data-value=""
                              >
                                <input
                                  aria-autocomplete="list"
                                  aria-describedby="react-select-2-placeholder"
                                  aria-expanded="false"
                                  aria-haspopup="true"
                                  aria-label="textbox"
                                  autocapitalize="none"
                                  autocomplete="off"
                                  autocorrect="off"
                                  class="react-select__input"
                                  id="react-select-2-input"
                                  role="combobox"
                                  spellcheck="false"
                                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                  tabindex="0"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                            >
                              <span
                                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                              />
                              <div
                                aria-hidden="true"
                                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="css-tj5bde-Svg"
                                  focusable="false"
                                  height="20"
                                  viewBox="0 0 20 20"
                                  width="20"
                                >
                                  <path
                                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="align-items-center pb-2 row"
                    >
                      <div
                        class="col-3"
                      >
                        <label
                          for="marketplace-product-category"
                        >
                          product_property_mappings_filter_marketplace_product_category_label
                        </label>
                      </div>
                      <div
                        class="col-6"
                      >
                        <div
                          class="react-select css-b62m3t-container"
                        >
                          <span
                            class="css-1f43avz-a11yText-A11yText"
                            id="react-select-3-live-region"
                          />
                          <span
                            aria-atomic="false"
                            aria-live="polite"
                            aria-relevant="additions text"
                            class="css-1f43avz-a11yText-A11yText"
                          />
                          <div
                            class="react-select__control css-13cymwt-control"
                          >
                            <div
                              class="react-select__value-container css-1fdsijx-ValueContainer"
                            >
                              <div
                                class="react-select__placeholder css-1jqq78o-placeholder"
                                id="react-select-3-placeholder"
                              >
                                product_property_mappings_filter_marketplace_product_category_placeHolder
                              </div>
                              <div
                                class="react-select__input-container css-qbdosj-Input"
                                data-value=""
                              >
                                <input
                                  aria-autocomplete="list"
                                  aria-describedby="react-select-3-placeholder"
                                  aria-expanded="false"
                                  aria-haspopup="true"
                                  aria-label="textbox"
                                  autocapitalize="none"
                                  autocomplete="off"
                                  autocorrect="off"
                                  class="react-select__input"
                                  id="react-select-3-input"
                                  role="combobox"
                                  spellcheck="false"
                                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                  tabindex="0"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                            >
                              <span
                                class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                              />
                              <div
                                aria-hidden="true"
                                class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="css-tj5bde-Svg"
                                  focusable="false"
                                  height="20"
                                  viewBox="0 0 20 20"
                                  width="20"
                                >
                                  <path
                                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="row"
                    >
                      <div
                        class="text-center col-10"
                      >
                        <button
                          class="me-2 btn btn-danger"
                          type="button"
                        >
                          common_button_reset_label
                        </button>
                        <button
                          class="btn btn-success"
                          type="button"
                        >
                          common_button_apply_label
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="p-datatable p-component p-datatable-selectable p-datatable-resizable p-datatable-resizable-fit p-datatable-responsive-scroll data-table-mh"
          data-scrollselectors=".p-datatable-wrapper"
          pr_id_1=""
        >
          <div
            class="p-datatable-header"
          >
            <div
              class="row"
            >
              <div
                class="d-flex justify-content-start align-items-center"
              >
                <div
                  class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
                >
                  <span
                    class="p-input-icon-right d-block"
                  >
                    <input
                      class="p-inputtext p-component"
                      placeholder="common_input_search_placeHolder"
                      value=""
                    />
                  </span>
                </div>
                <button
                  class="btn-h-95 btn btn-primary"
                  type="button"
                >
                  common_button_search_label
                </button>
              </div>
            </div>
          </div>
          <div
            class="p-datatable-wrapper"
          >
            <table
              class="p-datatable-table"
              role="table"
            >
              <thead
                class="p-datatable-thead"
              >
                <tr
                  role="row"
                >
                  <th
                    aria-sort="ascending"
                    class="p-sortable-column p-resizable-column p-highlight"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_property_mappings_table_header_column_marketplace_product_property
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-resizable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_property_mappings_table_header_column_is_mandatory
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    class="p-resizable-column"
                    role="columnheader"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_property_mappings_table_header_column_mms_product_property
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="p-datatable-tbody"
              >
                <tr
                  class="p-datatable-emptymessage"
                  role="row"
                >
                  <td
                    colspan="3"
                    role="cell"
                  >
                    product_property_mappings_table_empty_message
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="p-paginator p-component p-paginator-bottom table-paginator"
          >
            <div
              class="p-paginator-left-content"
            >
              <button
                aria-label="refresh-button"
                class="p-button p-component p-button-text p-button-icon-only"
                type="button"
              >
                <span
                  class="p-button-icon p-c pi pi-refresh"
                />
                <span
                  class="p-button-label p-c"
                >
                   
                </span>
              </button>
            </div>
          </div>
          <div
            class="p-column-resizer-helper"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="breadcrumb m-0"
        >
          <li
            class="breadcrumb-item"
          >
            <a
              href="/product-category-mappings"
            >
              <span>
                product_category_mappings_page_title
              </span>
            </a>
          </li>
          <li
            aria-current="page"
            class="breadcrumb-item active"
          >
            <span>
               
              product_property_mappings_page_title
               
            </span>
          </li>
        </ol>
      </nav>
      <div
        class="col"
      >
        <div
          class="page-title-box"
        >
          <h4
            class="page-title"
          >
            product_property_mappings_page_title
          </h4>
        </div>
      </div>
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="custom-accordion accordion"
              id="accordion"
            >
              <div
                class="d-flex align-items-center"
              >
                <span
                  class=""
                  style="margin-right: 0.5rem;"
                >
                  <span>
                    <div
                      class="d-flex align-items-start"
                    >
                      <i
                        class="mdi mdi-chevron-down d-none d-sm-inline-block align-middle"
                        style="transform: scale(1.5);"
                      />
                    </div>
                  </span>
                </span>
                <span
                  style="font-size: 1.2rem; margin: 0px;"
                >
                  product_property_mappings_filter_title
                </span>
              </div>
              <div
                class="accordion-collapse collapse show"
              >
                <div
                  class="px-0 container-fluid"
                >
                  <div
                    class="align-items-center pb-2 row"
                  >
                    <div
                      class="col-3"
                    >
                      <label
                        for="marketplace-type"
                      >
                        product_property_mappings_filter_marketplace_type_label
                      </label>
                    </div>
                    <div
                      class="col-6"
                    >
                      <div
                        class="react-select css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-2-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                        />
                        <div
                          class="react-select__control css-13cymwt-control"
                        >
                          <div
                            class="react-select__value-container css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1jqq78o-placeholder"
                              id="react-select-2-placeholder"
                            >
                              product_property_mappings_filter_marketplace_type_placeHolder
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-describedby="react-select-2-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                aria-label="textbox"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-2-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="align-items-center pb-2 row"
                  >
                    <div
                      class="col-3"
                    >
                      <label
                        for="marketplace-product-category"
                      >
                        product_property_mappings_filter_marketplace_product_category_label
                      </label>
                    </div>
                    <div
                      class="col-6"
                    >
                      <div
                        class="react-select css-b62m3t-container"
                      >
                        <span
                          class="css-1f43avz-a11yText-A11yText"
                          id="react-select-3-live-region"
                        />
                        <span
                          aria-atomic="false"
                          aria-live="polite"
                          aria-relevant="additions text"
                          class="css-1f43avz-a11yText-A11yText"
                        />
                        <div
                          class="react-select__control css-13cymwt-control"
                        >
                          <div
                            class="react-select__value-container css-1fdsijx-ValueContainer"
                          >
                            <div
                              class="react-select__placeholder css-1jqq78o-placeholder"
                              id="react-select-3-placeholder"
                            >
                              product_property_mappings_filter_marketplace_product_category_placeHolder
                            </div>
                            <div
                              class="react-select__input-container css-qbdosj-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-describedby="react-select-3-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                aria-label="textbox"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class="react-select__input"
                                id="react-select-3-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-tj5bde-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="text-center col-10"
                    >
                      <button
                        class="me-2 btn btn-danger"
                        type="button"
                      >
                        common_button_reset_label
                      </button>
                      <button
                        class="btn btn-success"
                        type="button"
                      >
                        common_button_apply_label
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-resizable p-datatable-resizable-fit p-datatable-responsive-scroll data-table-mh"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-start align-items-center"
            >
              <div
                class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
              >
                <span
                  class="p-input-icon-right d-block"
                >
                  <input
                    class="p-inputtext p-component"
                    placeholder="common_input_search_placeHolder"
                    value=""
                  />
                </span>
              </div>
              <button
                class="btn-h-95 btn btn-primary"
                type="button"
              >
                common_button_search_label
              </button>
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="ascending"
                  class="p-sortable-column p-resizable-column p-highlight"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_property_mappings_table_header_column_marketplace_product_property
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column p-resizable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_property_mappings_table_header_column_is_mandatory
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  class="p-resizable-column"
                  role="columnheader"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_property_mappings_table_header_column_mms_product_property
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="3"
                  role="cell"
                >
                  product_property_mappings_table_empty_message
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                 
              </span>
            </button>
          </div>
        </div>
        <div
          class="p-column-resizer-helper"
          style="display: none;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
